script/release-notes/notes.js

```diff
@@ -300,33 +303,50 @@ const checkCache = async (name, operation) => {
 
 const getPullRequest = async (number, owner, repo) => {
   const name = `${owner}-${repo}-pull-${number}`
+
   return checkCache(name, async () => {
-    try {
-      return await octokit.pulls.get({ number, owner, repo })
-    } catch (error) {
-      // Silently eat 404s.
-      // We can get a bad pull number if someone manually lists
-      // an issue number in PR number notation, e.g. 'fix: foo (#123)'
-      if (error.code !== 404) {
-        throw error
-      }
+    let failCount = 0
+    const tryGetPR = () => {
+      octokit.pulls.get({ number, owner, repo })
+        .then(result => {
+          return result
+        }).catch(error => {
+          if (failCount === MAX_FAIL_COUNT) {
+            // Silently eat 404s.
+            // We can get a bad pull number if someone manually lists
+            // an issue number in PR number notation, e.g. 'fix: foo (#123)'
+            if (error.code !== 404) throw error
+          } else {
+            failCount++
+            setTimeout(tryGetPR, CHECK_INTERVAL)
+          }
+        })
     }
+    tryGetPR()
   })
 }
 
 const getComments = async (number, owner, repo) => {
   const name = `${owner}-${repo}-pull-${number}-comments`
   return checkCache(name, async () => {
-    try {
-      return await octokit.issues.listComments({ number, owner, repo, per_page: 100 })
-    } catch (error) {
-      // Silently eat 404s.
-      // We can get a bad pull number if someone manually lists
-      // an issue number in PR number notation, e.g. 'fix: foo (#123)'
-      if (error.code !== 404) {
-        throw error
-      }
+    let failCount = 0
+    const tryGetComments = () => {
+      octokit.issues.listComments({ number, owner, repo, per_page: 100 })
```

Comment:
Same comment above about not having an await in play.  This video here explains a way to do retries on an async function: https://www.youtube.com/watch?v=7N8wbrCYqeM